File: python/lc202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        m = set()
        while n > 1:
            cur = n
            nxt = 0
            while cur > 0:
                rem = cur % 10
                nxt = nxt + rem * rem
                cur = cur // 10
            if nxt in m:
                return False
            else:
                m.add(nxt)
            n = nxt
        return n == 1

File: cs/lc202.cs
public class Solution {
    public bool IsHappy(int n) {
        var m = new HashSet<int>();
        while(n > 1)
        {
            int cur = n;
            int nxt = 0;
            while(cur > 0)
            {
                int rem = cur % 10;
                nxt = nxt + rem * rem;
                cur = cur / 10;
            }
            if(m.Contains(nxt))
                return false;
            else
                m.Add(nxt);
            n = nxt;
        }
        return n == 1;
    }
}
